### PR TITLE
Adding check-submit command

### PR DESCRIPTION
**Pull request by dmichaels-harvard**
This is a SubmitCGAP script which will check the status of a previously run submission. It is useful if the user either kills the submission command (e.g. submit-metadata or submit-ontology) before it is done waiting for the results or if the submission command times out waiting for the results. It is invoked with the submission (GUID) ID.

### CODE REVIEW

**Comment by netsettler** [comment]
@dmichaels-harvard It would be good to write a description of this PR. But in particular, I assume this is addressing [SubmitCGAP needs a 'check-status' command (C4-711)](https://hms-dbmi.atlassian.net/browse/C4-711), although I notice you picked a different name than check-status. Probably the name should be reviewed before we merge because I think it's very important to use nouns and verbs correctly. There is no such thing as "a submit". It is "a submission". And I think the word "status" is useful here. So maybe `check-submission-status` would be a good name?

I've also created [SubmitCGAP PR #40](https://github.com/dbmi-bgm/SubmitCGAP/pull/40), which is a layered PR that will fix the broken tests on this PR.

**Review by netsettler — APPROVED** [review]
After having made a number of test improvements and related fixes, this looks good to me.

**Review by aschroed — APPROVED** [review]
I have not had a chance to test this and not sure I will before I am away next week but the command looks useful and you should not let me hold this up.

File: submit_cgap/submission.py
Comment (drio18) [inline submit_cgap/submission.py:256]: I believe this should default to a list, though no functional difference. Similarly for line 281 for submission centers.
```suggestion
                     for consortium in user_record.get('consortia', [])]
```

File: submit_cgap/submission.py
Comment (drio18) [inline submit_cgap/submission.py:694]: A little confused at what's going on here. It doesn't look like the `app` is changed, so why call the function again with the same args? This seems to be occurring elsewhere as well, so likely I'm just not understanding something; consider refactoring into its own function with a separate docstring to explain the pattern.

File: submit_cgap/submission.py
Comment (drio18) [inline submit_cgap/submission.py:688]: Type annotations would be great to have for other code in this PR as well.

**Review by drio18 — APPROVED** [review]
One bigger picture question plus some minor comments on the code with which you can do as you see fit.

The larger questions is: Are we planning to use this repo for submissions to SMaHT? I know some additions have been made for limited, internal use with 4DN ontology submissions, but what's the plan for SMaHT? I assumed we would be creating a separate repo for it, but some of the additions here suggest we're planning to use this one.

File: submit_cgap/submission.py
Comment (netsettler) [reply submit_cgap/submission.py:256]: Yeah, since in either case it will do a null iteration, it hardly matters, but it will be more harmonious to use `[]`. :)

File: submit_cgap/submission.py
Comment (netsettler) [reply submit_cgap/submission.py:694]: But it's re-called in a context where the app can work properly. The important part of the re-call is the wrapper context manager.

File: submit_cgap/submission.py
Comment (netsettler) [reply submit_cgap/submission.py:688]: I'll look into adding some, though this is a work in progress.

**Pull request merged**
